### ISSUE TITLE
areComponentsEqual doesn't work?

### ISSUE THREAD

**Issue by Kimahriman** [gaearon/react-hot-loader#828]
### Description

I've been trying to implement a fix for an issue I had an react-bootstrap (https://github.com/react-bootstrap/react-bootstrap/issues/2963), but I can't seem to get `areComponentsEqual` to work like it says it should. In the simplest example, I took the `webpack` example and modified App.js to be

```javascript
import React from 'react'
import { hot, areComponentsEqual } from 'react-hot-loader'
import Counter from './Counter'

const App = () => {
  const element = <Counter/>
  console.log(areComponentsEqual(element.type, Counter))
  
  return (
    <h1>
      Hello, world.<br />
      <Counter />
    </h1>
  )
}

export default hot(module)(App)
```

### Expected behavior

`true` is printed in the console

### Actual behavior

`false` is printed in the console

Is there something I'm missing here or doing wrong?

### Environment

React Hot Loader version: 4.0.0-beta.18

Run these commands in the project folder and fill in their results:

1. `node -v`: v8.4.0
2. `npm -v`: 5.6.0

Then, specify:

1. Operating system: OSX Sierra
2. Browser and version: Chrome 63.0.3239.132

I can create a full demo repo but this seems straightforward.

**Comment by theKashey** [comment]
I am not sure why, but this function does not work this way. 🤷‍♂️
And the test we have checking how RHL can compare the old, and the new version of Component.

**Comment by neoziro** [comment]
Fixed in v4.0.0-beta.19.
